### DISCUSSION TITLE
ignore detections with low confidence or bounding boxes that are too small

### DISCUSSION THREAD

**Discussion by ghost** [AlexeyAB/darknet#8274] [Q&A]
Hi,
I am trying to detect parking spots and I want to ignore detections with bounding boxes that are too small (lets say I only want detections with width and height above 25 pixels) and detections with low confidence (lets say I only want detections with confidence higher than 0.5).
Is it possible to make these changes with hope to get better map results? if so, where I need to make the changes?

**Comment by stephanecharette** [comment]
You're looking at it the wrong way.  Don't change darknet to behave this way.  Instead, change your own application to ignore what you want.  So darknet gives out all these results, and your application would ignore the ones that are too small.  As for confidence level, this is a typical setting.  See "threshold".

**Reply by ghost** [reply]
regarding the confidence -
Im training with -map switch to get a graph of the map, so i just need to add -thresh 0.5 to the training command? 
is the confidence score is the same as  iou_thresh? because i think the network default iou_thresh is 0.5 but the lowest confidence im getting is 0.25.  
I saw that in the cfg file there are parameters named ignore_thresh =0.7 and truth_thresh=1 at each detection layer. should i change them also? what is their meaning?
regarding the size of bounding boxes -
Im already getting good results but small percentage of the detections are detections of half parking at the edge of the images and i wish to ignore them when calculating the map and im not sureabout how to do it. 
Thanks for your help and sorry for the confusion. I am a student and this my first experience with neural networks.

**Comment by ghost** [comment]
@stephanecharette

**Comment by stephanecharette** [comment]
No, you use threshold during inference, not training.  For training, I recommend using the script generated by DarkMark, or doing it as shown in this FAQ:  https://www.ccoderun.ca/programming/darknet_faq/#training_command

When it comes to ignoring certain predictions...your application is free to do whatever it wants with the results.  If a bbox is near the edge of the image, and the area (width x height) is < some magic value you determine works for you, then ignore it.

**Reply by ghost** [reply]
the thing is that because I'm training with -map option I'm doing inference every time a map calculation is being made(if I understand correctly) so I want these map calculations will take the thresh into account.

@stephanecharette
